src/shared/utils.ts

```diff
@@ -29,3 +29,13 @@ export function groupBy(array: any[], key: string) {
 
   return result;
 }
+
+/**
+ * Converts a Date object to a YYYY-MM-DD string (in UTC)
+ */
+export function toUtcDateString(date: Date) {
```

Comment:
not opposed to this because it's so small but we could throw a library in here and not maintain it `date-fns` is a good one if this repo does not already have one. 

Also, this is not a blocking comment.